DigitalLearningSolutions.Data.Tests/Services/EmailServiceTests.cs

```diff
@@ -0,0 +1,296 @@
+
+            // Then
+            A.CallTo(() =>
+                    smtpClient.Authenticate("username", "password", default(CancellationToken))
+                )
+                .MustHaveHappened();
+        }
+
+        [Test]
+        public void The_server_details_are_correct()
+        {
+            // When
+            emailService.SendEmail(testEmail);
+
+            // Then
+            A.CallTo(() =>
+                    smtpClient.Connect(
+                        "smtp.example.com",
+                        25,
+                        SecureSocketOptions.Auto,
+                        default(CancellationToken)
+                    )
+                )
+                .MustHaveHappened();
+        }
+
+        [Test]
+        public void The_sender_email_address_is_correct()
+        {
+            // When
+            emailService.SendEmail(testEmail);
+
+            // Then
+            A.CallTo(() =>
+                    smtpClient.Send(
+                        A<MimeMessage>.That.Matches(m =>
+                            m.From.ToString() == "test@example.com"
+                        ),
+                        default(CancellationToken),
+                        null
+                    )
+                )
+                .MustHaveHappened();
+        }
+
+        [Test]
+        public void The_email_subject_line_is_correct()
+        {
+            // When
+            emailService.SendEmail(testEmail);
+
+            // Then
+            A.CallTo(() =>
+                    smtpClient.Send(
+                        A<MimeMessage>.That.Matches(m =>
+                            m.Subject.ToString() == "Test Subject Line"
+                        ),
+                        default(CancellationToken),
+                        null
+                    )
+                )
+                .MustHaveHappened();
+        }
+
+        [Test]
+        public void The_email_text_body_is_correct()
+        {
+            // When
+            emailService.SendEmail(testEmail);
+
+            // Then
+            A.CallTo(() =>
+                    smtpClient.Send(
+                        A<MimeMessage>.That.Matches(m =>
+                            m.TextBody.ToString() == "Test body"
+                        ),
+                        default(CancellationToken),
+                        null
+                    )
+                )
+                .MustHaveHappened();
+        }
+
+        [Test]
+        public void The_email_HTML_body_is_correct()
+        {
+            // When
+            emailService.SendEmail(testEmail);
+
+            // Then
+            A.CallTo(() =>
+                    smtpClient.Send(
+                        A<MimeMessage>.That.Matches(m =>
+                            m.HtmlBody.ToString() == "<body style= 'font - family: Calibri; font - size: small;'>\r\n" +
+                            "                                    <p>Test Body</p>\r\n" +
```

Comment:
Hmm.. This is a bit odd (having all that spacing hardcoded) - could we pull the HTML string into a private variable that can be referenced here too?